GitExtUtils/Validates.cs

```diff
@@ -12,13 +12,15 @@ internal static class Validates
         // Extensions.
 
         [DebuggerStepThrough]
-        public static void NotNull<T>([ValidatedNotNull, NotNull] T? value)
+        public static T NotNull<T>([ValidatedNotNull, NotNull] T? value)
             where T : class
         {
             if (value == null)
             {
                 Fail("Value must not be null.");
             }
+
+            return value;
```

Comment:
To be honest, this change makes me very uneasy. This method is designed to be either no-op or terminate.

Reply:
Have a look at the usage, please. Having extra lines for `NotNull` feels non-fluent to me and requires additional identifiers, which cause unnecessary conflicts in this case.
https://github.com/gitextensions/gitextensions/pull/11442/files#diff-2c88663149829a858d94b4984e40320868afe1509d4af42fbc2a8833cd5cc901